### ISSUE TITLE
object-literal-sort-keys does not handle spreads well

### ISSUE THREAD

**Issue by giladgray** [palantir/tslint#2554]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: VSCode & gulp-tslint

#### TypeScript code being linted

```ts
const options: Tether.ITetherOptions = {
        constraints: useSmartPositioning ? [DEFAULT_CONSTRAINTS] : undefined,
        ...tetherOptions,
        attachment: getPopoverAttachment(position),
        bodyElement: fakeHtmlElement,
        classPrefix: "pt-tether",
        element,
        target,
        targetAttachment: getTargetAttachment(position),
    };
```

#### Actual behavior

`constraints` is marked as being out of order, wants it to appear after `classPrefix` (classic aphabetization), but this is _not equivalent code_ since `constraints` appears in its own object before `tetherOptions`.

#### Expected behavior

spreads should break up properties into separate groups (cuz they're compiled to separate objects in `__assign` helper)

```js
// compiled tsc output, modified for readability (and only key names)
var options = tslib_1.__assign(
    { constraints }, 
    tetherOptions, 
    { attachment, bodyElement, classPrefix, element, target, targetAttachment },
);
```
